apps/builder/app/builder/features/style-panel/sections/backgrounds/background-parser.ts

```diff
@@ -0,0 +1,120 @@
+import * as csstree from "css-tree";
+import { colord } from "colord";
+import { parseCssValue } from "../../shared/parse-css-value";
+import type {
+  InvalidValue,
+  LayersValue,
+  RgbValue,
+  UnparsedValue,
+} from "@webstudio-is/css-data";
+
+export const gradientNames = [
+  "conic-gradient",
+  "linear-gradient",
+  "radial-gradient",
+  "repeating-conic-gradient",
+  "repeating-linear-gradient",
+  "repeating-radial-gradient",
+];
+
+export const parseBackground = (
+  background: string
+): {
+  backgroundImage: LayersValue | InvalidValue;
+  backgroundColor: RgbValue | undefined;
+} => {
+  const layers: string[] = [];
+
+  let tokenStream = background.trim();
+
+  tokenStream = tokenStream.endsWith(";")
+    ? tokenStream.slice(0, -1)
+    : tokenStream;
+
+  const cleanupKeywords = ["background:", "background-image:"];
+
+  for (const cleanupKeyword of cleanupKeywords) {
+    tokenStream = tokenStream.startsWith(cleanupKeyword)
+      ? tokenStream.slice(cleanupKeyword.length).trim()
+      : tokenStream;
+  }
+
+  const cssAst = csstree.parse(tokenStream, { context: "value" });
+
+  let backgroundColorRaw: string | undefined;
+
+  let callDepth = 0;
+
+  csstree.walk(cssAst, {
+    enter: (node, item, list) => {
+      if (node.type === "Function") {
+        if (gradientNames.includes(node.name)) {
+          layers.push(csstree.generate(node));
+        }
+
+        // If the depth is at level 0 and the next item is null, it's likely that the backgroundColor
+        // is written as rgba(x,y,z,a) or similar format.
+        if (item.next === null && callDepth === 0) {
```

Comment:
I don't like callDepth logic very much, seems fragile, but I don't fully understand the thing